django-stubs/db/models/fields/related_descriptors.pyi

```diff
@@ -136,7 +141,7 @@ class ManyRelatedManager(BaseManager[_M], Generic[_M]):
     async def aset(self, objs: QuerySet[_M] | Iterable[_M | int], *, bulk: bool = ..., clear: bool = ...) -> None: ...
     def clear(self) -> None: ...
     async def aclear(self) -> None: ...
-    def __call__(self, *, manager: str) -> ManyRelatedManager[_M]: ...
+    def __call__(self, *, manager: str) -> Self: ...
```

Comment:
This seems suspect to me. Looking at the code, it resolves another manager from the Model and creates a ManyRelatedManager from it.

AFAICT they may be different manager classes and even have different base classes. So the previous annotation would be more accurate.

Reply:
Yeah, either isn't good enough. Getting this right is a whole other ticket (and then some).

I can't recall why I changed it to be honest.

I'll revert it.

Reply:
@intgr totally forgot about this. I've reverted just now

Reply:
No problem. Sorry, I haven't had time to review this thoroughly yet.